src/backend/distributed/commands/table.c

```diff
@@ -163,7 +162,8 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 			LockColocationId(cacheEntry->colocationId, ShareLock);
 		}
 
-		/* invalidate foreign key cache if the table involved in any foreign key */
+		/*
```

Comment:
I'm kind of wondering about the PreprocessDropTableStmt function. Is it always guaranteed to be called when a table is dropped? (e.g. for drop schema?)

anything that breaks if it isn't?

Reply:
> is it always guaranteed to be called when a table is dropped? (e.g. for drop schema?)

I don't think so